pkg/dom0-ztools/rootfs/etc/init.d/000-kexec

```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
+
+#
+# Kexec setup before everything else.
+#
+if cat /proc/cmdline | grep -q crashkernel; then
+	#
+	# Don't load kernel if no `crashkernel=Y@X` cmdline.
+	# This means we are on arch which does not support kdump
+	# (e.g. riscv64) or xen (see pkg/grub/rootfs.cfg)
+	#
+	/usr/sbin/kexec -p --reuse-cmdline --command-line="init 1 irqpoll maxcpus=1 reset_devices" /boot/kernel
+elif test -f /proc/vmcore; then
+	#
+	# We are in dump capture kernel, nice. Remount /etc as rw to
+	# stop further containerd execution.
+	#
```

Comment:
I don't know if @deitch or @rvs has ideas for other ways to make linuxkit not run the servic containers.

Reply:
The containers - onboot (runc), services (containerd), onshutdown(runc) - are handled via `linuxkit/init`, specifically [here](https://github.com/linuxkit/linuxkit/blob/45429986290f3811b6da093e8fdbbaffce96616e/pkg/init/cmd/service/system_init.go#L61). That in turn is called as part of `service containerd start`, which is called by `rc.init` [here](https://github.com/linuxkit/linuxkit/blob/45429986290f3811b6da093e8fdbbaffce96616e/pkg/init/cmd/rc.init/main.go#L424).

The actual entry that starts `containerd` is installed as part of `pkg/containerd` [here](https://github.com/linuxkit/linuxkit/blob/45429986290f3811b6da093e8fdbbaffce96616e/pkg/containerd/Dockerfile#L9).

Those don't currently have any way to disable containerd, but it isn't too hard to override it by having a later layer remove the `init.d` link if you want to build an image without containerd (or just remove that entry), or, if you want it to be conditional, having an earlier layer remove it on boot.

Reply:
Can I use "layers" without starting the containerd on early boot? I need to run a memory dump asap, when there is a bare kernel and no services running. Collecting of a dump  is a conditional operation - I collect dumps in a capture kernel, when original kernel crashed. 

I also can follow your suggestion and enable layer with the containerd init file further during boot (once I detect there is nothing to dump and kernel has not been crashed), but again the same problem: can I play with "layers" if no containerd service is around?